refactor(createRoutes): simplify and DRY routes and remove comments

devopsmariocom opened the pull request.

devopsmariocom committed:
refactor(createRoutes): simplify and DRY routes and remove comments

Hotell commented on src/routes.tsx:
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
-        const importModules = Promise.all( [
-          System.import( './faq/Faq' ),
-        ] );
+        importModules.catch( ( err: any ) => {
```
Chain

Hotell commented on src/routes.tsx:
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
-          renderRoute( component.About );
+        importModules.then( ( [
+          component ] ) => {
+          renderRoute( component[ Object.keys( component )[0] ] );
```
I don't like this

devopsmariocom replied:
me neither any suggestions?

Hotell commented on src/routes.tsx:
```diff
@@ -19,67 +14,31 @@ export function createRoutes( /*store*/ ): RouteConfig[] {
       getComponent( nextState, cb ) {
         const importModules = Promise.all( [
-          System.import( './about/About' ),
+          System.import( `./${nextState.location.pathname.slice(1)}/index` ),
```
Btw this Promise.all is unnecessary in this case, we are loading just one file/module. Root one

devopsmariocom committed:
refactor(createRoutes): remove redundant Promise.all

devopsmariocom committed:
refactor(createRoutes): add removeLeadingSlash for readability

Hotell commented on src/routes.tsx:
```diff
@@ -1,85 +1,40 @@
-const errorLoading = ( err: any ) => {
-  console.error( 'Dynamic page loading failed', err );
-};
+import { NotFound } from './not-found/NotFound';
```
we are importing from barrel, not from file directly

Hotell commented on src/routes.tsx:
```diff
@@ -1,85 +1,40 @@
   cb( null, componentModule );
 };
 
+const getObjectFirstKey = ( object ) => Object.keys( object )[0];
```
no types, what is object?

Hotell commented on src/routes.tsx:
```diff
@@ -1,85 +1,40 @@
 
+const getObjectFirstKey = ( object ) => Object.keys( object )[0];
+
+const removeLeadingSlash = ( routePath ) => routePath.slice( 1 );
```
no types, what is routePath? if string, consider `substring` `slice` is more supposed to be used with Array

devopsmariocom committed:
refactor(createRoutes): add ComponentModule type and use proper substring

devopsmariocom commented on src/routes.tsx:
```diff
@@ -1,85 +1,40 @@
-        const renderRoute = loadModule( cb );
+        System.import( `./${ removeLeadingSlash( nextState.location.pathname ) }/index` )
+          .then( ( component ) => {
+            loadModule( cb )( component[ getObjectFirstKey( component ) ] );
```
still it more feels like default exports make sense for page components WDYT?

see https://github.com/ngParty/React-scaffold/pull/18

Hotell commented on src/routes.tsx:
```diff
@@ -3,15 +3,21 @@ import { hashHistory, Router, RouteConfig } from 'react-router';
-import { NotFound } from './not-found/NotFound';
+import { NotFound } from './not-found';
+import StatelessComponent = React.StatelessComponent;
+import Component = React.Component;
```
This namespace aliasing is wrong. Pls watch what webstorm generates for your next time before committing

devopsmariocom committed:
refactor(createRoutes): leverage default exports (#18)

* refactor(createRoutes): leverage default exports

to simplify module loading logic

* refactor(createRoutes): set any for incompatible and types prevent implicit any

type definitions seems tobe off by a lot, so let's go with any before we or somebody else will fix the types

devopsmariocom commented:
merging reason: we have to move on, any changes to current state can be and will be easily reverted
